### ISSUE TITLE
InstanceNorm does not catch dim mismatch

### ISSUE THREAD

**Issue by osadj** [pytorch/pytorch#109652]
### 🐛 Describe the bug

InstanceNorm with a specified input C works on tensors of arbitrary dimension (with the number of channels not necessarily matching C) without throwing a dimension mismatch error. For example:

```
m = torch.nn.InstanceNorm1d(64)
input = torch.randn(4, 80, 300)
output = m(input)
```

### Versions

PyTorch v2.0

**Comment by mikaylagawarecki** [comment]
This was already fixed on `main` to throw a warning by https://github.com/pytorch/pytorch/pull/94624/files#diff-d49c3b3b20727c953a6f14cba34746844c3e617ec4a9016a398919430d766491, the fix should be available in the next release (2.1.0).

https://github.com/pytorch/pytorch/blob/567e8ebf947deccf093d5340540595e8321814ae/torch/nn/modules/instancenorm.py#L73-L82


```python
m = torch.nn.InstanceNorm1d(64)
input = torch.randn(4, 80, 300)
output = m(input)
UserWarning: input's size at dim=1 does not match num_features. You can silence this warning by not passing in num_features, which is not used because affine=False
```

`num_features` is only used to determine the size of `weight` and `bias` if `affine=True`  in which case an error is thrown (`affine=False` is the default for LayerNorm however)

I think it is inaccurate that the warning can be silenced by not passing in `num_features` (since `num_featrues` is a required argument), so perhaps it is actionable to reword that part of warning

**Comment by fzimmermann89** [comment]
It would be nicer to actually support what the warning suggests and allow features=1 or features=None to be used with affine=False. It is annoying that one has to specify the number of features just to avoid the warning
